File: android/src/main/kotlin/ir/aligator/appmarket/markets/Jhoobin.kt
package ir.aligator.appmarket.markets

import android.app.Application
import android.content.ComponentName
import android.content.ContentUris
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.provider.CalendarContract
import ir.aligator.appmarket.exceptions.FeatureNotSupportedException

class Jhoobin(private val context: Context) : AppMarket {
    override val id: Int = 3
    override val name: String = "ایده پرداز ژوبین"

    override fun showAppPage(packageId: String) {
        val intent = Intent(Intent.ACTION_VIEW)
        intent.data = Uri.parse("jhoobin://search?q=$packageId")
        context.startActivity(intent)
    }

    override fun showAppComments(packageId: String) {
        val intent = Intent(Intent.ACTION_VIEW)
        intent.data = Uri.parse("jhoobin://comment?q=$packageId")
        context.startActivity(intent)
    }

    override fun showDeveloperApps(devId: String) {
        val intent = Intent(Intent.ACTION_VIEW)
        intent.data = Uri.parse("jhoobin://collection?type=APP&id=$devId")
        context.startActivity(intent)
    }

    override fun getVersionCode(packageId: String): Long {
        throw FeatureNotSupportedException(name, "getVersionCode")
    }
}

File: lib/src/jhoobin.dart
import 'package:android_intent_plus/android_intent.dart';
import 'package:iran_appmarket/iran_appmarket.dart';

class Jhoobin implements IAppMarket {

  @override
  void showAppPage(String packageName) {
    AndroidIntent(
      action: "android.intent.action.VIEW",
      data: 'jhoobin://search?q=$packageName',
    ).launch();
  }

  @override
  void showAppComments(String packageName) {
    AndroidIntent(
      action: "android.intent.action.VIEW",
      data: 'jhoobin://comment?q=$packageName',
    ).launch();
  }


  @override
  void showDeveloperApps(String developerId) {
    AndroidIntent(
      action: "android.intent.action.VIEW",
      data: 'jhoobin://collection?type=APP&id=$developerId',
    ).launch();
  }

}